### ISSUE TITLE
Hot change max_payload in server config, but client not receive INFO_ARG message

### ISSUE THREAD

**Issue by taiheserver** [nats-io/nats-server#4200]
i found nats only send INFO_ARG to client when client reconnect or server cluster had changed.
i used simple nats version, not jetstream

**Comment by derekcollison** [comment]
Or when a server enters lame duck mode as well. That is correct for now.

**Comment by taiheserver** [comment]
But if server  enters lame duck mode, it will stop.
If i use jetstream, can I achieve hot update max_payload？

**Comment by derekcollison** [comment]
LameDuck will eventually exit but the client will have time to take action. We also plan on doing intelligent reconnect at some point for clients.

In terms of max_payload, what are you trying to use it for? In general there is little reason to change this value in the majority of systems.

**Comment by taiheserver** [comment]
When our system running online, we found some message too large to over max_payload, but we didn't want to stop the system. 
I set the max_payload to 6M, but after we updated, there was a message that it reached 7M, which was not tested by our test before going online.
Then I signal SIGHUP to the server that I want to be able to hot-update max_payload，I sent the signal to all the nodes of the server cluster, but nats client didn't receive the configuration change. So I had to restart my service for the nats client to get the latest config again.
Although changing max_payload is not a frequent operation, it still happens occasionally
nats server version: 2.8.4
nats client is nats.go: 1.25.0
